nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -22,12 +22,18 @@ let
 
   kernel = config.boot.kernelPackages;
 
-  splKernelPkg = kernel.spl;
-  zfsKernelPkg = kernel.zfs;
-  zfsUserPkg = pkgs.zfs;
+  packages = if lib.versionAtLeast kernel.kernel.version "4.9" then {
```

Comment:
I don't think we should install an unstable version of ZFS automatically and without warning if a stable Linux kernel is to be installed.